test/integration/builder/additional.js

```diff
@@ -256,36 +256,50 @@ module.exports = function(knex) {
     });
 
 
-    it('.timeout() should throw TimeoutError', function() {
+    it('.timeout() should throw TimeoutError', function(done) {
       var dialect = knex.client.config.dialect;
-      var rows = _.times(dialect === 'sqlite3' ? 100 : 3000, function() {
-        return {
-          id: 1,
-          value: 'rO8F8YrFS6uoivuRiVnwrO8F8YrFS6uoivuRiVnwuoivuRiVnw'
-        };
-      });
+      if(dialect === 'sqlite3') { done(); } //TODO -- No built-in support for sleeps
+      var testQueries = {
+        'postgres': function() {
+          return knex.raw('SELECT pg_sleep(5)');
+        },
+        'mysql': function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        'mysql2': function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        maria: function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        mssql: function() {
+          return knex.raw('WAITFOR DELAY \'00:00:05\'');
+        },
+        oracle: function() {
+          return knex.raw('dbms_lock.sleep(5)');
+        },
+        'strong-oracle': function() {
+          return knex.raw('dbms_lock.sleep(5)');
+        }
+      };
 
-      return knex.schema.dropTableIfExists('TimeoutTest')
-        .then(function() {
-          return knex.schema.createTable('TimeoutTest', function(table) {
-            table.integer('id');
-            table.string('value', 50);
-          });
-        })
-        .then(function() {
-          return knex('TimeoutTest')
-            .insert(rows)
-            .timeout(1);
-        })
+      if(!testQueries.hasOwnProperty(dialect)) {
+        throw new Error('Missing test query for dialect: ' + dialect);
+      }
+
+      var query = testQueries[dialect]();
+
+      return query.timeout(1000)
```

Comment:
Perhaps use a much shorter timeout here? Tests are synchronous and having ones that make pauses for timeout detection gets real irritating real fast.